devtools/server/actors/replay/connection.js

```diff
@@ -518,7 +531,21 @@ function updateRecordingState(key, state) {
   const timestamps = current && current.timestamps || {};
   timestamps[state] = Date.now();
 
-  return recordings.set(key, { state, timestamps });
+  const entry = {
+    state,
+    recording: current ? current.recording : null,
+    timestamps
+  };
+  recordings.set(key, entry);
+  return entry;
+}
+
+function addRecordingInstance(key, recording) {
+  const entry = recordings.get(key);
+  if (!entry) {
+    entry = updateRecordingState(key, RecordingState.RECORDING);
+  }
+  entry.recording = recording;
```

Comment:
Should this assert that `entry.state === RecordingState.RECORDING` here?

Reply:
I don't think it technically matters in this case, we just want to make sure it's marked RECORDING if there isn't an entry already.